content/en/blog/_posts/Use containerd to handle k8s.gcr.io deprecation

```diff
@@ -0,0 +1,174 @@
+server = "https://k8s.gcr.io"
+
+[host."https://registry.k8s.io"]
+capabilities = ["pull", "resolve"]
+EOF
+```
+
+Image pull requests to `k8s.gcr.io` will now be sent to `registry.k8s.io`.
+
+Restart containerd and kubelet for the change to take effect.
+
+```bash
+systemctl restart containerd kubelet
+```
+
+Let’s validate that images are indeed getting pulled from the new registry. I added an entry to my `/etc/hosts` file to break `k8s.gcr.io`:
+
+![Image.png](https://res.craft.do/user/full/9d54cc03-adfe-f72f-3389-565eb7356d1d/doc/13FE78CC-1DDD-47D9-B875-EE3FB15F916E/8AEBA792-CA42-4C8B-9C8D-E34AC55A9847_2/7A4cAKtUBNGVqmUysXRwpW0WGy3cNGz14xxujKlsmc8z/Image.png)
+
+Conatinerd can no longer pull an image from `k8s.gcr.io`
+
+![Image.jpeg](https://res.craft.do/user/full/9d54cc03-adfe-f72f-3389-565eb7356d1d/doc/13FE78CC-1DDD-47D9-B875-EE3FB15F916E/D382C5B0-437D-44E3-BAA2-E32032FF7E29_2/zmbCw5wl0LyV1TNJdiEQxCnSqKBS3s4IDZWYljKaLvkz/Image.jpeg)
+
+I can tell `ctr` to use the mirror by specifying the `—hosts-dir` parameter:
+
+```bash
+ctr images pull --hosts-dir "/etc/containerd/certs.d" k8s.gcr.io/pause:latest
+```
+
+This time the operation succeeds.
+
+![Image.jpeg](https://res.craft.do/user/full/9d54cc03-adfe-f72f-3389-565eb7356d1d/doc/13FE78CC-1DDD-47D9-B875-EE3FB15F916E/94C5856D-E54D-414D-86E6-5B7C70A7989B_2/2bTbUm39iI52DAfS4D9GgH1T1LW5pBbWxTHNIjgS2v4z/Image.jpeg)
+
+Any Pods I create now onwards will use the new registry even though the manifests reference old registry. Here’s a test using a pause container.
+
+```bash
+kubectl create deployment pause --image k8s.gcr.io/pause
+```
+
+![Image.jpeg](https://res.craft.do/user/full/9d54cc03-adfe-f72f-3389-565eb7356d1d/doc/13FE78CC-1DDD-47D9-B875-EE3FB15F916E/88014DF3-E73E-4A60-B881-64C31B8210B6_2/lnuzXKg7G4RwURvQUxrZwRFuBDljKWpX6m01nVydnxoz/Image.jpeg)
+
+Perfect! Kubernetes could create Pods even though I blocked `k8s.gcr.io` on the node.
+
+## What’s the best way to implement this in production?
+
+In my little demo, I changed a single node in the cluster. What about the rest of the nodes?
+
+There are three ways you can use to implement this change on every node in your cluster:
+
+1. The easiest way is to use a daemonset to change to containerd config.toml and add hosts.toml file. IBM cloud has shared [this](https://raw.githubusercontent.com/IBM-Cloud/kube-samples/master/containerd-registry-daemonset-example) on Github
+2. You can use [EC2 user data](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html) or [AWS Systems Manager](https://aws.amazon.com/systems-manager/) to make this change when a node gets created
+3. You can [create your own AM](https://docs.aws.amazon.com/eks/latest/userguide/eks-ami-build-scripts.html)I
+
+## What if I use Docker as runtime?
+
+Starting Kubernetes version `1.24`, containerd is the only runtime available in Amazon EKS AMIs. If you have an edge case that requires using Docker, there's still hope.
```

Comment:
Kubernetes version 1.24 (and later releases) don't include direct support for Docker as a container
runtime, though you can make it work with a shim. If you require using Docker directly, there's still hope.
```